sys/hashes/sha1.c

```diff
@@ -0,0 +1,134 @@
+/* This code is public-domain - it is based on libcrypt
```

Comment:
Could you also add a doxygen header into this file? It might be of interest later who imported it and so on... Thx